src/js/js.c

```diff
@@ -21,6 +21,137 @@ static void reportError(JSContext *js_context, const char *message, JSErrorRepor
 			(unsigned int) report->lineno + 1, message);
 }
 
+// debugging purpose
+// to be deleted
+void js_dump_response(JSContext *cx, jsval v) {
+	jsval rval;
+	
+	if (JS_GetProperty(cx, JSVAL_TO_OBJECT(v), "code", &rval)) {
+		printf("code = ");
+		js_dump_value(cx, rval);
+	}
+
+	if (JS_GetProperty(cx, JSVAL_TO_OBJECT(v), "message", &rval)) {
+		printf("message = ");
+		js_dump_value(cx, rval);
+	}
+	
+	if (JS_GetProperty(cx, JSVAL_TO_OBJECT(v), "disconnect", &rval)) {
+		printf("disconnect = ");
+		js_dump_value(cx, rval);
+	}
+}
+
+int js_get_code(jsval v) {
+	jsval code;
+
+	if (JS_GetProperty(js_context, JSVAL_TO_OBJECT(v), "code", &code)) {
+		printf("cod=%d\n", JSVAL_TO_INT(code));
+		return JSVAL_TO_INT(code);
+	}
+
+	return -1;
+}
+
+char* js_get_message(jsval v) {
+	jsval message;
+	char *c_str;
+
+	if (JS_GetProperty(js_context, JSVAL_TO_OBJECT(v), "message", &message)) {
+		c_str = JS_EncodeString(js_context, JSVAL_TO_STRING(message));
+		return c_str;
+	}
+
+	return NULL;
+}
+
+int js_get_disconnect(jsval v) {
+	jsval disconnect;
+
+	if (JS_GetProperty(js_context, JSVAL_TO_OBJECT(v), "disconnect", &disconnect)) {
+		return JSVAL_TO_BOOLEAN(disconnect);
+	}
+
+	return 1;
```

Comment:
If the 'disconnect' property is not found, the session should be kept alive (i.e. return 0).

Rationale: if a JS handler "forgot" to set 'disconnect' in the returned object and you returned 'true', the client connection would be dropped immediately without any response to the client.